src/coreclr/vm/loaderallocator.cpp

```diff
@@ -427,7 +429,7 @@ LoaderAllocator * LoaderAllocator::GCLoaderAllocators_RemoveAssemblies(AppDomain
                     {
                         pLoaderAllocator->ClearMark();
                     }
-                    else if (!pLoaderAllocator->IsAlive())
+                    if (!pLoaderAllocator->IsAlive())
```

Comment:
This doesn't make sense. The fix looks like it says we can have a LoaderAllocator which is both Marked, and !IsAlive and that !Alive should trump the marking logic.